src/http/request.cr

```diff
@@ -98,6 +98,18 @@ class HTTP::Request
     value
   end
 
+  # Return request host from headers
+  def host
+    return unless @headers["Host"]
+    index = @headers["Host"].index(":")
+    index ? @headers["Host"][0...index] : @headers["Host"]
```

Comment:
Please assign `@headers["Host"]` to a local variable, otherwise we have 2 lookups inside that hash.